File: clients/client_tcp.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <winsock2.h>
#include <ws2tcpip.h>

#pragma comment(lib, "ws2_32.lib")

#define PORT 8080
#define BUFFER_SIZE 1024

int main() {
    system("chcp 65001 > nul");
    WSADATA wsaData;
    SOCKET sock = 0;
    struct sockaddr_in serv_addr;
    char *hello = "Hello from client";
    char buffer[BUFFER_SIZE] = {0};
    
    if (WSAStartup(MAKEWORD(2, 2), &wsaData) != 0) {
        printf("WSAStartup failed\n");
        return -1;
    }
    
    if ((sock = socket(AF_INET, SOCK_STREAM, 0)) == INVALID_SOCKET) {
        printf("Socket creation error: %d\n", WSAGetLastError());
        WSACleanup();
        return -1;
    }
    
    serv_addr.sin_family = AF_INET;
    serv_addr.sin_port = htons(PORT);
    
    if (inet_pton(AF_INET, "127.0.0.1", &serv_addr.sin_addr) <= 0) {
        printf("\nInvalid address/ Address not supported \n");
        closesocket(sock);
        WSACleanup();
        return -1;
    }
    
    if (connect(sock, (struct sockaddr *)&serv_addr, sizeof(serv_addr)) < 0) {
        printf("\nConnection Failed \n");
        closesocket(sock);
        WSACleanup();
        return -1;
    }
    
    send(sock, hello, strlen(hello), 0);
    printf("Hello message sent\n");
    
    int valread = recv(sock, buffer, BUFFER_SIZE - 1, 0);
    printf("Server response: %s\n", buffer);
    
    closesocket(sock);
    WSACleanup();
    
    return 0;
}

File: clients/client_tcp.py
import socket

def main():
    client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    
    try:
        server_address = ('localhost', 8080)
        print(f"Подключаюсь к серверу {server_address[0]}:{server_address[1]}")
        client_socket.connect(server_address)
        
        message = "Привет, сервер! Это тестовое сообщение от клиента."
        client_socket.sendall(message.encode('utf-8'))
        print(f"Отправлено сообщение: {message}")
        
        data = client_socket.recv(1024)
        print(f"Получен ответ от сервера: {data.decode('utf-8')}")
        
    except Exception as e:
        print(f"Ошибка: {e}")
    finally:
        client_socket.close()
        print("Соединение закрыто")

if __name__ == "__main__":
    main()